src/corelibs/U2View/src/ov_msa/exclude_list/MsaExcludeList.cpp

```diff
@@ -0,0 +1,593 @@
+
+    U2OpStatus2Log os;
+    Document *document = format->createNewLoadedDocument(ioAdapterFactory, savePath, os);
+    for (int i = 0; i < nameListView->count(); i++) {
+        QListWidgetItem *listItem = nameListView->item(i);
+        int excludeListRowId = listItem->data(LIST_ITEM_DATA_ROW_ID).toInt();
+        DNASequence sequence = sequenceByExcludeListRowId[excludeListRowId];
+        U2EntityRef ref = U2SequenceUtils::import(os, document->getDbiRef(), U2ObjectDbi::ROOT_FOLDER, sequence);
+        CHECK_OP(os, nullptr);
+        document->addObject(new U2SequenceObject(sequence.getName(), ref));
+    }
+
+    SAFE_POINT_OP(os, nullptr);
+    saveTask = new SaveDocumentTask(document);
+    connect(new TaskSignalMapper(saveTask), &TaskSignalMapper::si_taskFinished, this, [this] { saveTask = nullptr; });
+    AppContext::getTaskScheduler()->registerTopLevelTask(saveTask);
+    updateState();
+    return saveTask;
+}
+
+void MsaExcludeListWidget::saveExcludeFileToNewLocation() {
+    QString dir = GUrl(excludeListFilePath).dirPath();
+    QString newFilePath = U2FileDialog::getSaveFileName(this,
+                                                        tr("Select new exclude list file name"),
+                                                        dir,
+                                                        FileFilters::createFileFilter(tr("Exclude list FASTA file"), {EXCLUDE_LIST_FILE_SUFFIX}));
+    CHECK(!newFilePath.isEmpty() && newFilePath != excludeListFilePath, );
+    if (!newFilePath.endsWith(EXCLUDE_LIST_FILE_SUFFIX)) {
+        newFilePath = newFilePath + "." + EXCLUDE_LIST_FILE_SUFFIX;
+    }
+    if (!FileAndDirectoryUtils::canWriteToPath(newFilePath)) {
+        QMessageBox::critical(this, L10N::errorTitle(), tr("File is not writable: %1").arg(newFilePath));
+        return;
+    };
+    if (auto task = runSaveTask(newFilePath)) {
+        connect(new TaskSignalMapper(task), &TaskSignalMapper::si_taskSucceeded, this, [this, newFilePath] {
+            excludeListFilePath = newFilePath;
+            isDirty = false;
+        });
+    }
+}
+
+void MsaExcludeListWidget::handleLoadTaskStateChange() {
+    SAFE_POINT(loadTask != nullptr, "Load task is nullptr in handleLoadTaskStateChange", );
+    CHECK(loadTask->isFinished(), );
+
+    auto task = loadTask;
+    loadTask = nullptr;
+
+    if (task->hasError()) {
+        stateLabel->setText(tr("Error loading exclude list file: ").arg(task->getError()));
+    } else if (!task->isCanceled()) {
+        QList<GObject *> objects = task->getDocument()->findGObjectByType(GObjectTypes::SEQUENCE);
+        nameListView->clear();
+        U2OpStatus2Log os;
+        for (auto object : qAsConst(objects)) {
+            auto sequenceObject = qobject_cast<U2SequenceObject *>(object);
+            SAFE_POINT(sequenceObject != nullptr, "Not a sequence object: " + object->getGObjectName(), );
```

Comment:
`L10n`

Reply:
L10n in SAFE_POINT will not pass code review.